### ISSUE TITLE
ALOpsDockerCreate - the bcartifacts.cache/Sandbox is filling up on drive C

### ISSUE THREAD

**Issue by JesperNavtilus** [HodorNV/ALOps#201]
Problem:
ALOpsDockerCreate fills up the c: drive on our local pipeline agent server with cached sandbox artifacts.  The C drive is only meant for the OS.

After installing the Azure Pipeline Agent on drive D instead of drive C, the OnPrem artifacts are being cached on the D drive, but the Sandbox artifacts are being cached on the C drive.

Solution:
When ALOpsDockerCreate use (I guess) the Download-Artifacts method in NavContainerHelper it should set the -basePath (which is default c:\bcartifacts.cache) to use the same drive as the Azure Pipeline Agents '_work' folder is on. Or do what ever is seems to do right for the OnPrem artifacts.

**Comment by waldo1001** [comment]
We don't really have a solution for now (we don't use download-artifacts, by the way).

But .. you could use mklink.  Just create the folder on the d-drive, and create an mklink on the c-drive to this folder, and that kind of will do that for you.. .

More info:
https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/mklink

**Comment by JesperNavtilus** [comment]
"Ok" - the mklink works (thank you) - for now, but I hope that you will address this issue again so that that ALOps wrapping of NavContainerHelper/Container/PowerShell/OS cmd works more indifferent on the artifacttype (OnPrem/Sandbox) being used.

**Comment by waldo1001** [comment]
Yes, caching is on our list - but it's more complicated than it seems ;-).

**Comment by fvet** [comment]
@waldo1001 I remember having read somewhere a retention setting would exist to remove unused bcartifacts.cache files (7 - 10 days?). With the frequent nextMinor / nextMajor build, our C:\bcartifacts.cache is also piling up with unused artifacts. Any options in ALOps to monitor the cache contents?

**Comment by AdminHodor** [comment]
Dear @fvet , @JesperNavtilus ,

We just released v1.435.1856 which should tackle this issue. We have a new task [ALOpsAgentMaintenance] which allows you cleanup the agent.
Please note it could take a couple of pipeline runs for DevOps to pickup the new version.

```
- task: ALOpsAgentMaintenance@1
  displayName: 'ALOps Agent Maintenance'
  inputs:
    removeoldtasks: true
    removeunusedcontainers: true
    removebcartifacts: true
    daysunused: 14
```

Kind regards,
